cli/internal/helm/versionlister.go

```diff
@@ -0,0 +1,145 @@
+/*
+Copyright (c) Edgeless Systems GmbH
+
+SPDX-License-Identifier: AGPL-3.0-only
+*/
+
+package helm
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/edgelesssys/constellation/v2/internal/semver"
+	"helm.sh/helm/v3/pkg/action"
+	"helm.sh/helm/v3/pkg/release"
+	"k8s.io/client-go/util/retry"
+)
+
+// ReleaseVersionLister can list the versions of a helm release.
+type ReleaseVersionLister interface {
+	CurrentVersion(release string) (semver.Semver, error)
+}
+
+// ReleaseVersionClient is a client that can retrieve the version of a helm release.
+type ReleaseVersionClient struct {
+	config *action.Configuration
+}
+
+// NewReleaseVersionClient creates a new ReleaseVersionClient.
+func NewReleaseVersionClient(kubeConfigPath string, log debugLog) (*ReleaseVersionClient, error) {
+	config, err := newActionConfig(kubeConfigPath, log)
+	if err != nil {
+		return nil, err
+	}
+	return &ReleaseVersionClient{
+		config: config,
+	}, nil
+}
+
+// listAction execute a List action by wrapping helm's action package.
+// It creates the action, runs it at returns results and errors.
+func (c ReleaseVersionClient) listAction(release string) (res []*release.Release, err error) {
+	action := action.NewList(c.config)
+	action.Filter = release
+	// during init, the kube API might not yet be reachable, so we retry
+	err = retry.OnError(retry.DefaultBackoff, func(err error) bool {
+		return err != nil
+	}, func() error {
+		res, err = action.Run()
+		return err
+	})
+	return
+}
+
+// CurrentVersion returns the version of the currently installed helm release.
+func (c ReleaseVersionClient) CurrentVersion(release string) (semver.Semver, error) {
```

Comment:
```suggestion
// currentVersion returns the version of the currently installed helm release.
func (c ReleaseVersionClient) currentVersion(release string) (semver.Semver, error) {
```
Can be kept private as it is only used in the `helm` package